### PR TITLE
releng: Add jobs for the release-1.31 branch

### PR DESCRIPTION
**Pull request by mehabhalodiya**
This PR manually adds the changes for the release-1.31 branch.

/assign @dims @xmudrii 
cc @kubernetes/release-engineering

/sig release
/area release-eng
/priority critical-urgent

ref: https://github.com/kubernetes/sig-release/issues/2577

### CODE REVIEW

File: releng/test_config.yaml [hunk 1]
```diff
@@ -76,13 +76,6 @@ jobs:
     interval: 2h
     sigOwners: [sig-cloud-provider-gcp]
     releaseBlocking: true
-  ci-kubernetes-e2e-gce-cos-k8sstable1-default:
-    args:
-    - --env=ENABLE_POD_SECURITY_POLICY=true
-    interval: 2h
-    sigOwners: [sig-cloud-provider-gcp]
-    releaseBlocking: true
-    testgridNumFailuresToAlert: 6
```
Comment (xmudrii) [inline releng/test_config.yaml:85]: Why did you delete this job?

Comment (mehabhalodiya) [reply releng/test_config.yaml:85]: Sorry if I misunderstood this comment: https://github.com/kubernetes/test-infra/issues/32340#issuecomment-2260548829

From that don't we agree to remove it? I apologize if I get it wrong. Let me know if we need that, and I can add it back. Thanks!

Comment (xmudrii) [reply releng/test_config.yaml:85]: The comment was referring to the `env` flag, the flag should be removed, but the job should stay in place.

Comment (mehabhalodiya) [reply releng/test_config.yaml:85]: Oh sorry, I see. Let me fix it.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml [hunk 2]
```diff
@@ -254,13 +254,13 @@ periodics:
   - 'perfDashPrefix: kubemark-500Nodes-1.27'
   - 'perfDashJobType: performance'
 - annotations:
-    fork-per-release-cron: 0 8-20/24 * * *
+    fork-per-release-cron: ""
```
Comment (dims) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml:257]: Looks like this is getting dropped, is that ok?

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml [hunk 3]
```diff
@@ -79,7 +79,7 @@ periodics:
           cpu: "1"
           memory: 3Gi
 - annotations:
-    fork-per-release-periodic-interval: 24h
+    fork-per-release-periodic-interval: ""
```
Comment (dims) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml:82]: this is getting dropped... intentional? or bug?

Comment (dims) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml:82]: is it because this is the oldest stable branch?

**Comment by dims** [comment]
Looks like needs some touch up!
```
Gubernator configuration file is out of sync!
33d32
<   - pull-kubernetes-e2e-gce-ubuntu-containerd
44d42
<   - pull-kubernetes-verify-govet-levee
Run `gubernator/update_config.sh`
```

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.31.yaml [hunk 4]
```diff
@@ -1681,10 +2240,108 @@ presubmits:
...
+      repo: windows-testing
+      workdir: true
+    labels:
+      preset-azure-community: "true"
+      preset-capz-containerd-1-7-latest: "true"
+      preset-capz-windows-2022: "true"
+      preset-capz-windows-common-pull: "true"
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    name: pull-kubernetes-e2e-capz-windows-master
```
Comment (dims) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.31.yaml:2319]: mismatch with the context in line 2296 above?

```

2/484 Tests Failed. | expand_less
-- | --
k8s.io/test-infra/config/tests/jobs: TestContextMatches expand_less0s{Failed  === RUN   TestContextMatches     jobs_test.go:263: Context does not match job name: pull-kubernetes-e2e-capz-windows-master in kubernetes/kubernetes --- FAIL: TestContextMatches (0.00s) } | k8s.io/test-infra/config/tests/jobs: TestContextMatches expand_less | 0s | {Failed  === RUN   TestContextMatches     jobs_test.go:263: Context does not match job name: pull-kubernetes-e2e-capz-windows-master in kubernetes/kubernetes --- FAIL: TestContextMatches (0.00s) }
k8s.io/test-infra/config/tests/jobs: TestContextMatches expand_less | 0s
{Failed  === RUN   TestContextMatches     jobs_test.go:263: Context does not match job name: pull-kubernetes-e2e-capz-windows-master in kubernetes/kubernetes --- FAIL: TestContextMatches (0.00s) }


```

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml (hunk 3 above)
Comment (xmudrii) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml:82]: > is it because this is the oldest stable branch?

Yes, exactly! This annotation contains a list of annotations to be used when forking and rotating this job. Because this job is not getting forked/rotated after this release, it doesn't have any interval left.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml (hunk 2 above)
Comment (xmudrii) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml:257]: https://github.com/kubernetes/test-infra/pull/33201#discussion_r1701796786

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.31.yaml (hunk 4 above)
Comment (mehabhalodiya) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.31.yaml]: Referring https://github.com/kubernetes/test-infra/pull/31294#issuecomment-1822103763 can we remove this job?

**Comment by mehabhalodiya** [comment]
Also, I see
```
could not write config: 4 errors occurred:
	* configuration error for (Dashboard) sig-release-1.26-blocking: contains no tabs
	* configuration error for (Dashboard) sig-release-1.26-informing: contains no tabs
	* could not find the referenced (TestGroup) pull-kubernetes-verify-govet-levee
	* could not find the referenced (TestGroup) pull-kubernetes-e2e-gce-ubuntu-containerd
FAIL	k8s.io/test-infra/config/tests/testgrids	33.849s
```
Should I remove 1.26 testgrids/dashboard? or do I understand this error wrong?

**Comment by dims** [comment]
/approve
/lgtm

thanks @mehabhalodiya
